prime-router/docs/design/design/deduplication/deduplication.md

```diff
@@ -0,0 +1,197 @@
+
+# Universal Pipeline Deduplication
```

Comment:
Consider adding a new section at the bottom that has the proposed convert step SRD changes so those can be reviewed as well. If these changes are only to the convert step, then I would copy the convert step requirements, amend them, and label each new or changed req with "(NEW)" tag.

Reply:
Added Convert step changes under "Proposed Universal Pipeline SRD Additions", please review those when you have time.

Also noticed that there is some related language under 
```
Submission Step Business Logic Requirements 
    6.4. The submission step shall NOT accept the other custom HTTP Headers currently defined in RequestFunction.kt:
        6.4.4. AllowDuplicate – no clear requirements for this in UP at this time
```

This might also be a @jsutantio question. From my recollection we've been pretty adamant about not allowing senders to avoid this workflow on a per request basis. If that is still true then I would want to update this description to reflect that.

Reply:
Spoke with @jsutantio after standup today and she confirmed that in the UP we do not want to enable this HTTP param which could override the sender settings for duplication. I have pushed a commit to add this section to proposed UP SRD changes:
```
6.4. [No Change] The submission step shall NOT accept the other custom HTTP Headers currently defined in RequestFunction.kt: 
    6.4.4. [UPDATED] AllowDuplicate – Not to be used for UP De-duplication
```